### DISCUSSION TITLE
enable scheduling extension (beyond pg_cron)

### DISCUSSION THREAD

**Discussion by jonathanstanley** [supabase/supabase#16250] [Feature Requests]
I know Supabase has [pg_cron](https://supabase.com/docs/guides/database/extensions/pg_cron) which can be used to create recurring schedules.

But is there an option for scheduling a function to occur ONCE (at a specific future time)? For example
- [pgagent](https://www.pgadmin.org/docs/pgadmin4/development/pgagent.html )
- [pg_timetable](https://github.com/cybertec-postgresql/pg_timetable)

I haven't used either of these, however `pg_timetable` looks very nice. For example ["self-destructive" jobs](https://www.cybertec-postgresql.com/en/pg_timetable-advanced-postgresql-job-scheduling/) can be set to run once (such as to terminate a free trial at a given date).

I realize technically this could be done with pg_cron by having the cron removal as part of the action. however, it would be nice to use regular dates (eg ISO) for one-off date scheduling. `schedule("2022-09-27 18:00:00.000",()=>{//do stuff})` and not have to convert a date to a cron format.

**Comment by pierroo** [comment]
Did you find any solution to that @jonathanstanley ?
I am currently searching for exactly that :)

Even for something as silly as "do this in 30 seconds"; 
I even considered making it an edge function that would just wait 30 seconds but... useless cost

**Reply by jonathanstanley** [reply]
using various workarounds for now 😕

**Comment by itisnajim** [comment]
did u tried to unschedule after a pg_cron job run your schedule (a function that contains `unschedule`) ?
`select cron.unschedule('webhook-every-minute'); -- pass the name of the cron job`
like in the blog https://supabase.com/blog/postgres-as-a-cron-server

**Comment by Dioxymore** [comment]
Any news about that? I was considering Supabase for a game, but I need to invoke functions at precise times down to the second (the game is turn-based, and I must be able to end a turn and update the state of the game something like "every 45 seconds"), and the current pg_cron doesn't seem to be adapted... 
Am I missing something or is it not possible with Supabase? Are there some known workarounds?

**Reply by GaryAustin1** [reply]
This seems like a Postgres question and not specific to Supabase.   The cron extension now has 1 second resolution https://github.com/citusdata/pg_cron .
But normally the extension is intended to run a repeating task on an interval.

**Reply by Dioxymore** [reply]
I tried that, but it seems to only be available since pg_cron 1.5, while the version in supabase is 1.4-1. But maybe that's a free-tier limitation I'm not aware of?

_"But normally the extension is intended to run a repeating task on an interval."_
I agree, but that was the point of the original post to ask for the implementation of other extensions beyond pg_cron that could be used in those use-cases =)

**Reply by GaryAustin1** [reply]
It is available if you update your Postgres version under infrastructure settings.
